src/controllers/secret.go

```diff
@@ -0,0 +1,194 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dana-team/platform-backend/src/types"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretController interface {
+	CreateSecret(types.CreateSecretRequest) (types.CreateSecretResponse, error)
+	GetSecrets(types.GetSecretsRequest) (types.GetSecretsResponse, error)
+	GetSecret(types.GetSecretRequest) (types.GetSecretResponse, error)
+	PatcnSecret(types.UpdateSecretRequest) (types.UpdateSecretResponse, error)
+	DeleteSecret(types.DeleteSecretRequest) (types.DeleteSecretResponse, error)
+}
+
+type secretController struct {
+	client *kubernetes.Clientset
+	ctx    context.Context
+	logger *zap.Logger
+}
+
+func NewSecretController(client *kubernetes.Clientset, context context.Context, logger *zap.Logger) SecretController {
+	return &secretController{
+		client: client,
+		ctx:    context,
+		logger: logger,
+	}
+}
+
+func (n *secretController) CreateSecret(request types.CreateSecretRequest) (types.CreateSecretResponse, error) {
+	n.logger.Debug(fmt.Sprintf("Trying to create secret: %q", request.Name))
+
+	var secretType v1.SecretType
+	response := types.CreateSecretResponse{}
+	name := request.Name
+	namespace := request.Namespace
+	data := make(map[string][]byte)
+
+	if request.Type == "tls" {
+		if request.Cert == "" || request.Key == "" {
+			return response, fmt.Errorf("cert and key are required for TLS secrets")
+		}
+		secretType = v1.SecretTypeTLS
+		data["tls.crt"] = []byte(request.Cert)
+		data["tls.key"] = []byte(request.Key)
+	} else if request.Type == "opaque" {
+		secretType = v1.SecretTypeOpaque
+		for _, kv := range request.Data {
+			data[kv.Key] = []byte(kv.Value)
```

Comment:
Does the K8S client take care of transforming the values to `base64`? Assume the frontend would send the values as regular non-decoded strings. CMIIW @arielsepton 

If it doesn't then we need to turn the values to `base64` before creating/patching the secret values.